### ISSUE TITLE
Overly complex code, possible bug?

### ISSUE THREAD

**Issue by d-frey** [PowerDNS/pdns#11549]
https://github.com/PowerDNS/pdns/blob/33c36defeb8fc9013b6b9942e4ad791ec67d65bb/modules/bindbackend/bindbackend2.cc#L1209-L1219

If I'm not mistaken, the above code contains two cases which lead to the exact same outcome. Both 1210 and 1219 both set `d_list` to `false` and the effect of 1211 is the same as 1215-1216 as in 1211 we know that `range.first == range.second`. So either the code from 1209-1219 can be simplifed to:

```c++
d_handle.d_list = false;
d_handle.d_iter = range.first;
d_handle.d_end_iter = range.second;
```

or there is a bug somewhere and the code is supposed to do something different in one of the cases.

**Comment by rgacogne** [comment]
I can't spot a bug, the condition is properly checked in `Bind2Backend::handle::get_normal()` later so I think the code has just been overly complicated since 76faefcfb7ae99d70da9019e5008954cab99d809 and could be simplified.

**Comment by d-frey** [comment]
Ok, do you want me to send a PR or would you like to handle it yourself?

**Comment by rgacogne** [comment]
If you don't mind opening a PR I believe we will gladly merge it :)

**Comment by Habbie** [comment]
Closed in #11552, thanks!
